sbrk.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#define SIZE_GiB		(1ULL << 30)
+#define SIZE_TiB		(1ULL << 40)
+
+/* 2 ^ 47 = 256 TiB, split in half for user/kernel address space. */
+#define MAX_USER_VM_4LVL	(128ULL * SIZE_TiB)
+
+int main()
+{
+	int ret;
+	int i=0;
+	void *ptr;
+
+	for (;;)
+	{
+		ptr = sbrk(1 * SIZE_GiB);
+		if (ptr == -1)
```

Comment:
sbrk.c:20:11: warning: comparison between pointer and integer
   if (ptr == -1)

You'll need to cast -1 in order to silence this warning, ie,  == (void *) -1